### PR TITLE
Fixes #34 - Implement News Feed Search

### LINKED ISSUE #34: Implement the NewsFeed Search
Implement the search in the News Feed. There is already code for the search, but it's doing nothing

### PR DESCRIPTION
**Pull request by rajasone**

### CODE REVIEW

File: ICT4DNews/app/src/main/res/menu/ict4dnews_menu.xml [hunk 1]
```diff
@@ -7,13 +7,13 @@
         android:icon="@drawable/ic_search_white_24dp"
         android:title="@string/search"
         app:actionViewClass="android.support.v7.widget.SearchView"
-        app:showAsAction="collapseActionView|ifRoom" />
```
Comment (spipau) [inline ICT4DNews/app/src/main/res/menu/ict4dnews_menu.xml]: Please always use `ifRoom` and never `always`, even Android Studio shows a warning 😉

File: ICT4DNews/app/src/main/res/menu/ict4dnews_menu.xml [hunk 2]
```diff
@@ -7,13 +7,13 @@
...
+        app:showAsAction="collapseActionView|always" />
 
     <item
         android:id="@+id/menu_filter"
         android:icon="@drawable/ic_filter_list_white_24dp"
         android:title="@string/filter"
-        app:showAsAction="ifRoom" >
```
Comment (spipau) [inline ICT4DNews/app/src/main/res/menu/ict4dnews_menu.xml]: Same here

File: ICT4DNews/.idea/codeStyles/codeStyleConfig.xml [hunk 3]
```diff
@@ -1,5 +0,0 @@
-<component name="ProjectCodeStyleConfiguration">
```
Comment (spipau) [inline ICT4DNews/.idea/codeStyles/codeStyleConfig.xml]: Why did you delete this file?

File: ICT4DNews/app/src/main/java/at/ict4d/ict4dnews/screens/news/list/ICT4DNewsFragment.kt [hunk 4]
```diff
@@ -59,41 +60,77 @@ class ICT4DNewsFragment : BaseNavigationFragment<ICT4DNewsViewModel, FragmentIct
...
+            override fun onMenuItemActionCollapse(p0: MenuItem?): Boolean {
+                adapter.submitList(model.searchedNewsList.value)
+                model.searchQuery = null
+                enableRefreshMenuItem(true)
+                return true
+            }
+        })
+    }
+
+    private fun enableRefreshMenuItem(isEnable: Boolean) {
```
Comment (spipau) [inline ICT4DNews/app/src/main/java/at/ict4d/ict4dnews/screens/news/list/ICT4DNewsFragment.kt]: change to variable name to `isEnabled` or `enable` - otherwise it's not English 😉

File: ICT4DNews/app/src/main/res/menu/ict4dnews_menu.xml (hunk 1 above)
Comment (rajasone) [reply ICT4DNews/app/src/main/res/menu/ict4dnews_menu.xml]: it was `ifRoom,` but while using search when you change the orientation and close the search the items with `ifRoom` attribute are not visible anymore until the change of orientation again

File: ICT4DNews/.idea/codeStyles/codeStyleConfig.xml (hunk 3 above)
Comment (rajasone) [reply ICT4DNews/.idea/codeStyles/codeStyleConfig.xml]: it must be a mistake sorry for that

File: ICT4DNews/app/src/main/res/menu/ict4dnews_menu.xml (hunk 1 above)
Comment (spipau) [reply ICT4DNews/app/src/main/res/menu/ict4dnews_menu.xml]: I cannot reproduce this? Works on my side? Can you test that again or provide me a detailed step-by-step guide to reproduce the bug?

Comment (spipau) [reply ICT4DNews/app/src/main/res/menu/ict4dnews_menu.xml]: OK, please change it back to `always` but add a good comment for our future selves 😉
